### DISCUSSION TITLE
Nomad compatibility

### DISCUSSION THREAD

**Discussion by ttaghavi** [ofek/csi-gcs#170] [Q&A]
Has anyone tried to make this CSI work with Nomad?

Any feedback/experiences welcome =)

**Comment by ofek** [comment]
Link?

**Reply by ttaghavi** [reply]
Nomad storage csi plugin documentation can be found [here](https://developer.hashicorp.com/nomad/docs/concepts/plugins/csi).

Two Nomad example configs for existing CSI plugins:

* [hetzner csi plugin](https://github.com/hetznercloud/csi-driver/issues/330#issuecomment-1321985192)
* [s3 csi plugin](https://github.com/ctrox/csi-s3/issues/64#issuecomment-1479654256)

**Reply by ofek** [reply]
This plugin conforms to the spec so everything should work as expected

**Reply by ttaghavi** [reply]
I see, thanks @ofek.

Though what I really wanted to know is if anyone had a working example. But I'll give it a shot myself.

What I am unsure about, when looking at the documentation is:

* how does the docker image docker.io/ofekmeister/csi-gcs differentiate between controller and node mode (through args?)
* what non-default parameters are needed to bring up the service (`parameters` block in the nomad volume specification)

But I guess I'll apply the proverb: experimenting is sometimes better than studying =)
